### DISCUSSION TITLE
Copilot in VS Code only returns single line of code

### DISCUSSION THREAD

**Discussion by denisyilmaz** [community/community#44563] [Copilot Conversations]
### Select Topic Area

Bug

### Body

<img width="631" alt="image" src="https://user-images.githubusercontent.com/4189607/212842816-978abf83-694e-447c-8396-8c2a33b3c185.png">

i disabled all extensions except Copilot but the error persists.

VSCode 1.74.3 (Universal) on MacBook Pro M1 Max

**Comment by Akash1134** [comment]
Hi @denisyilmaz We are reviewing this issue. From our product team member [kimadeline](https://github.com/kimadeline) as posted in this [discussion](https://github.com/orgs/community/discussions/40522#discussioncomment-4706865):

- Quick update on this issue:
- Despite the symptoms being similar, this is not related to the [multiline regression issue](https://github.com/orgs/community/discussions/13303) that happened around April-May 2022
- We are actively looking into it 🔍
- It seems like it should mostly an issue for code written in JS, TS, Go and Python (for all editors). If you encounter these issues in languages other than the ones listed there please speak up 📣

**Comment by bvanderdrift** [comment]
Curious to hear if there's any updates? Also dealing with this in VSCode in a TS project.

Seems some starts of lines only give a 1-line suggestion (like `export const`)

<img width="481" alt="image" src="https://user-images.githubusercontent.com/6398452/215739059-8db50c67-d5c5-4d29-91ba-ec28ff99b0c3.png">

whereas others (`functions ...` seems to work) does suggest multi-line. (Very odd suggestion though, it's post-transpiled TS code)

<img width="487" alt="image" src="https://user-images.githubusercontent.com/6398452/215738836-02abfc5c-1784-45d4-9aa0-e38bf5df1781.png">

**Reply by denisyilmaz** [reply]
Here it magically worked again since last week. But not 100% of the time.. would be great if this feature would be more predictable, specially for a paid feature…

**Comment by arielpollack** [comment]
having the same issue. I have to click "Tab" and "Enter" for it to complete a full function for example.

That's makes copilot not functional at all.

<img width="435" alt="image" src="https://user-images.githubusercontent.com/1862153/218996399-714e69ed-30a4-41eb-8548-50cb32e92c1d.png">

**Reply by DanielGit4** [reply]
I have the same issue in with copilot in VS-Code. This is actually a little disappointing for a paid product.

**Comment by DanielGit4** [comment]
are there any updates on this?

**Comment by ThiagoMaia1** [comment]
I'm experiencing the same. It seemed to be fixed a while back, but it's back to happening for the last few weeks

**Comment by denisyilmaz** [comment]
Same issue here. getting single line suggestions a lot of times. But i realized when opening GitHub Code Pilot (CTRL + Enter) i get full suggestions:

Suggesion for comment on previous line (hidden) "add arrow function to say "Hello {name}", as you can see, only single line is suggested:
<img width="621" alt="image" src="https://user-images.githubusercontent.com/4189607/222499053-31127445-0e7d-420e-ae02-1650cd5745cb.png">

Suggestioned in opened GitHub Copilot tab (has multiple lines, but misses the closing `}` on the function:
<img width="396" alt="image" src="https://user-images.githubusercontent.com/4189607/222499336-b3e1f6d3-74ec-4085-b270-156bf8db781d.png">

As a paid extension this is really disappointing…

**Comment by ThiagoMaia1** [comment]
@denisyilmaz How do you open that? I can't see that menu

**Reply by denisyilmaz** [reply]
on a mac just use keyboard shortcut `ctrl + enter` or via mouse click the three dots next to the GitHub Copilot tooltip in VS Code:

<img width="430" alt="image" src="https://user-images.githubusercontent.com/4189607/222505086-afed02cd-f632-4c8f-a3ac-dcc8cca3c0c7.png">

I use the latest version of VS Code with the GitHub Copilot extension installed.

**Reply by ThiagoMaia1** [reply]
Ah, sorry, I used command + enter, instead. Control + enter does work for me.

**Comment by cr1of0x** [comment]
GitHub Copilot Nightly extension works better for me and suggests all code inside function (except for the first line of function declaration).

<img width="358" alt="image" src="https://github.com/community/community/assets/87279678/9d58ba9c-0a22-469c-b4a4-5b4f4557087d">

**Comment by ahoopes16** [comment]
As of June this is still a problem for me

**Comment by sean256** [comment]
This is anecdotal, but of the 4 projects I use this with, it's much more common with my largest workspace. My smallest project always works.

**Comment by ddmuzyk** [comment]
I'm having the same issue, multiple lines are only happening with functions (I'm using javascript).

**Comment by samd1993** [comment]
Same problem here. It is just mildly infuriating enough that I had to come here looking for fix because sometimes I don't mind this to help me inspect all the generated code. But some lines are so repetitive that this becomes an issue

**Comment by HasanMothaffar-OSG** [comment]
Hey did anyone find a solution to this?
